packages/ember-application/tests/system/dependency_injection/to_string_test.js

```diff
@@ -1,62 +1,63 @@
 import { guidFor } from 'ember-utils';
-import { ENV, context } from 'ember-environment'; // lookup, etc
-import { run } from 'ember-metal';
-import Application from '../../../system/application';
+import { context } from 'ember-environment'; // lookup, etc
 import { Object as EmberObject } from 'ember-runtime';
-import DefaultResolver from '../../../system/resolver';
+
+import { moduleFor, AutobootApplicationTestCase, ModuleBasedTestResolver } from 'internal-test-helpers';
 
 let originalLookup, App;
 
-QUnit.module('Ember.Application Dependency Injection – toString', {
-  setup() {
+moduleFor('Ember.Application Dependency Injection - toString', class extends AutobootApplicationTestCase {
```

Comment:
Can this test use the `ApplicationTestCase` instead of the `AutobootApplicationTestCase`? Something like:

```js
moduleFor('Ember.Application Dependency Injection - DefaultResolver#toString', class extends ApplicationTestCase {
  constructor() {
    this._originalLookup = context.lookup;
    super();
    this.application.Post = EmberObject.extend();
    context.lookup = { App: this.application };
    return this.visit('/');
  }
  get applicationOptions {
    return assign(super.applicationOptions, {
      Resolver: DefaultResolver
    });
  }

  teardown() {
    super.teardown();
    context.lookup = originalLookup;
  }

  ['@test factories'](assert) {
    let PostFactory = this.applicationInstance.factoryFor('model:post').class;
    assert.equal(
      PostFactory.toString(), 'App.Post',
      'expecting the model to be post'
    );
  }
```

I'm actually pretty sure you can take out the `context` stuff entirely. With the default resolver the application itself is already the namespace. But I would need to muck with it to be sure.